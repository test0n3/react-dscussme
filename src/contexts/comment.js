import React from "react";

const CommentContext = React.createContext();

function CommentProvider({ comments, setComments, children }) {
  return (
    <CommentContext.Provider value={{ comments, setComments }}>
      {children}
    </CommentContext.Provider>
  );
}

function useComment() {
  return React.useContext(CommentContext);
}

export { CommentProvider, useComment };
