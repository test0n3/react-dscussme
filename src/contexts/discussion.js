import React from "react";

const DiscussionContext = React.createContext();

function DiscussionProvider({ discussions, setDiscussions, children }) {
  return (
    <DiscussionContext.Provider value={{ discussions, setDiscussions }}>
      {children}
    </DiscussionContext.Provider>
  );
}

function useDiscussion() {
  return React.useContext(DiscussionContext);
}

export { DiscussionProvider, useDiscussion };
