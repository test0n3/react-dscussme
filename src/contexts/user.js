import React from "react";

const UserContext = React.createContext();

function UserProvider({ user, setUser, children }) {
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
}

function useUser() {
  return React.useContext(UserContext);
}

export { UserProvider, useUser };
