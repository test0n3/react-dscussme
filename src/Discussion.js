/** @jsx jsx */
import React from "react";
import Comment from "./Comment";
import ButtonCreateComment from "./ButtonCreateComment";
import { jsx } from "@emotion/core";
import { Link } from "@reach/router";
import { useDiscussion } from "./contexts/discussion";
import { useUser } from "./contexts/user";

function Discussion({ getAllComments, id }) {
  const { discussions, setDiscussions } = useDiscussion();
  const { user } = useUser();
  const discussion = discussions.find(discussion => discussion.id == id);

  let comments = getAllComments(discussion.id);
  console.log("GAA", id);
  const discussionStyle = {
    fontSize: "1.2em",
    border: "1px solid #666666",
    borderRadius: "10px",
    margin: 10,
    padding: "0 10px"
  };

  return (
    <div css={{ margin: "70px 0" }}>
      <article css={{ ...discussionStyle }}>
        <section>
          <h3>{discussion.title}</h3>
          <p>{discussion.body}</p>
        </section>
        <section
          css={{
            display: "flex",
            flexDirection: "row",
            margin: 0,
            padding: 0,
            fontSize: "0.6em",
            justifyContent: "space-between"
          }}
        >
          <div css={{ display: "flex", flexDirection: "row" }}>
            <p css={{ margin: 5 }}>by {discussion.author.username}</p>
            <p css={{ margin: 5 }}>{discussion.date}</p>
          </div>
          <ButtonCreateComment />
        </section>
      </article>
      {/* <p>{discussion.id}</p> */}
      <article>
        <h2 css={{ marginLeft: 40 }}>Comments</h2>
        {comments.map(comment => {
          return <Comment key={comment.id} comment={comment} />;
        })}
      </article>
    </div>
  );
}

export default Discussion;
