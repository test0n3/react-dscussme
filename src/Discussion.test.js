import React from "react";
import { render } from "@testing-library/react";
import Discussion from "./Discussion";
import { DiscussionProvider } from "./contexts/discussion";
import { UserProvider } from "./contexts/user";

test("Discussion component", async () => {
  const getAllComments = () => [];
  const setDiscussions = jest.fn();
  
  const discussions = [
    {
      id: new Date("2019-06-13T21:50:18.315Z").getTime(),
      date: "2019-06-13T21:50:18.315Z",
      author: "Clarissa",
      title: "La vida es bella",
      body: "ideas sueltas"
    }
  ];

  const { asFragment } = render(
    <DiscussionProvider
      discussions={discussions}
      setDiscussions={setDiscussions}
    >
      <UserProvider>
        <Discussion
          getAllComments={getAllComments}
          id={new Date("2019-06-13T21:50:18.315Z").getTime()}
        />
      </UserProvider>
    </DiscussionProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
