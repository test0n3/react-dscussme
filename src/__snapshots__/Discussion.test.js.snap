// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Discussion component 1`] = `
<DocumentFragment>
  <p>
    1560462618315
  </p>
  <p>
    2019-06-13T21:50:18.315Z
  </p>
  <p />
  <p>
    La vida es bella
  </p>
  <p>
    ideas sueltas
  </p>
  <hr />
  <h2>
    Comments
  </h2>
  <button>
    <div
      style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
    >
      Create Comment
    </div>
    <span
      aria-hidden="true"
    >
      +
    </span>
  </button>
  <hr />
</DocumentFragment>
`;
